app/routers/wallet.py

```diff
@@ -16,15 +18,106 @@
 
 admin_url = os.getenv("ACAPY_ADMIN_URL")
 admin_port = os.getenv("ACAPY_ADMIN_PORT")
+# TODO Should the admin_api_key be a dummy variable so the controller doesn't function w/o providing it?
+# This all smells really - this has to be done in a better manner
 admin_api_key = os.getenv("ACAPY_ADMIN_API_KEY")
 is_multitenant = os.getenv("IS_MULTITENANT", False)
 ledger_url = os.getenv("LEDGER_NETWORK_URL")
 
 
-@router.get(
-    "/create-pub-did", tags=["did"], response_model=DidCreationResponse
-)
-async def create_public_did():
+async def __create_did(controller):
+    generate_did_res = await controller.wallet.create_did()
+    if not generate_did_res["result"]:
```

Comment:
So I'd really like to see if we can move this validation code somewhere else... 
Either we:
1. go through some analysis and decide how important it actually is - on a case by case basis and thus we can potentially remove it in some cases.
2. remove it altogether but make sure we throw the best error we can at the end (i.e. we make sure that our api responds with something potentially relevant). Error handling is _hard_ I know.
3. embed it in the wallet controller. This would be first prize though not sure the cloudcontroller people would be keen. They haven't done it so far so not sure they'll do it now
4. Create our own wrapper or something else round the controllers to centralise and manage this validation code. 

Number 4 is arguably the most viable and it's not a great deal of work. Just a lot of typing.